spec/requests/api/v1/participant_outcome_spec.rb

```diff
@@ -0,0 +1,92 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe "participant outcomes endpoint spec", :with_default_schedules, type: :request do
+  let(:token)                { LeadProviderApiToken.create_with_random_token!(cpd_lead_provider: provider) }
+  let(:bearer_token)         { "Bearer #{token}" }
+  let(:fake_logger)          { double("logger", info: nil) }
+  let(:provider) { create :cpd_lead_provider, :with_npq_lead_provider }
+  let(:npq_application) { create :npq_application, :accepted, npq_lead_provider: provider.npq_lead_provider }
+  let(:declaration) { create :npq_participant_declaration, participant_profile: npq_application.profile, cpd_lead_provider: provider }
+
+  describe "JSON Index Api" do
+    context "when authorized" do
+      before do
+        default_headers[:Authorization] = bearer_token
+        default_headers[:CONTENT_TYPE] = "application/json"
+      end
+
+      context "when no outcome exists" do
+        it "returns empty array" do
+          get "/api/v1/participants/npq/outcomes"
+          expect(response.status).to eq 200
+
+          expect(parsed_response).to eq("data" => [])
+        end
+      end
+
+      context "when outcome exists" do
+        let(:expected_response) do
+          expected_json_response(outcome:, profile: npq_application.profile)
+        end
+        let!(:outcome) { create :participant_outcome, participant_declaration: declaration }
+
+        it "returns serialised data" do
+          get "/api/v1/participants/npq/outcomes"
+          expect(response.status).to eq 200
+
+          expect(parsed_response).to eq(expected_response)
```

Comment:
I wonder if instead of retesting the exact values appear here (as we do the serialiser) we can have higher level tests such as checking the type is correct, the ids are correct and the number is correct. Similar for the paginations. An example of this would be: https://github.com/DFE-Digital/early-careers-framework/blob/develop/spec/requests/api/v3/statements_spec.rb#L57
We were hoping to eventually move this to some shared context as we are doing this across the API. Let me know you're thoughts

Reply:
I'd rather be explicit - this leaves no room for any typos in the controller - I think it this case the test for only certain aspects of the response would be more complex and harder to read.